### PR TITLE
Product Category API

### PR DESCRIPTION
**Pull request by NalineeS**

### CODE REVIEW

File: src/main/java/com/ecommerce/controller/ProductCategoryController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		return new ResponseEntity<>(productCategoryService.findProductCategoryById(categoryId), HttpStatus.OK);
+	}
+
+	@DeleteMapping("/delete/{categoryId}")
+	public ResponseEntity<?> deleteProductCategoryById(@PathVariable Integer categoryId) {
+		productCategoryService.deleteProductCategoryById(categoryId);
+		return new ResponseEntity<>(HttpStatus.OK);
+	}
+
+	@PutMapping("/edit/{categoryId}")
```
Comment (maheshsm) [inline src/main/java/com/ecommerce/controller/ProductCategoryController.java]: replace edit with update word

File: src/main/java/com/ecommerce/controller/ProductCategoryController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	@PostMapping("/")
+	public ResponseEntity<?> addNewProductCategory(@RequestBody ProductCategory productCategory) {
+
+		productCategoryService.addProductCategory(productCategory);
+
+		return new ResponseEntity<>(HttpStatus.CREATED);
+	}
+
+	@GetMapping("/getall")
```
Comment (maheshsm) [inline src/main/java/com/ecommerce/controller/ProductCategoryController.java]: get/all

File: src/main/java/com/ecommerce/repository/ProductCategoryRepository.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	 * To find Product Category By Category Id
+	 * 
+	 * @param categoryId
+	 */
+	public void deleteProductCategoryById(Integer categoryId) {
+		jdbcTemplate.update(SqlConstant.DELETE_PRODUCT_CATEGORY_BY_ID, categoryId);
+	}
+
+	/**
+	 *
```
Comment (maheshsm) [inline src/main/java/com/ecommerce/repository/ProductCategoryRepository.java:66]: add proper comments

File: src/main/java/com/ecommerce/service/ProductCategoryService.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.List;
+
+import com.ecommerce.model.ProductCategory;
+
+/**
+ * This is specification class for category activities.
+ */
+public interface ProductCategoryService {
+
+	Integer addProductCategory(ProductCategory productCategory);
```
Comment (maheshsm) [inline src/main/java/com/ecommerce/service/ProductCategoryService.java:18]: add comments on all methods

**Review by Nandu-Chopade — APPROVED** [review]
ProductCategory feature added to this master branch

**Pull request merged**